### PR TITLE
documentation:S3C-4144 milestone-1 validation

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
- Added Validation for [Milestone-1](https://github.com/scality/vmware-ose-scality/blob/development/1.0/docs/milestone-1-requirements.md)
- Updated README.md with steps to generate .p12 file with self-signed certificate

### CODE REVIEW

File: docs/milestone-1-validation.md
Comment (anurag4DSB) [inline docs/milestone-1-validation.md:8]: What do we mean by expose URL and port?

File: docs/milestone-1-validation.md
Comment (anurag4DSB) [inline docs/milestone-1-validation.md:12]: The application.properties might need a comment that we need to use superadmin creds there and not just any AK/SK. But is still fine.

File: docs/milestone-1-validation.md
Comment (anurag4DSB) [inline docs/milestone-1-validation.md:8]: Reference to the respective documentation is needed,
Also, what do we exactly mean by `expose the URL and port`?

File: docs/milestone-1-validation.md
Comment (anurag4DSB) [inline docs/milestone-1-validation.md:16]: Reference to documentation will be helpful

**Review by anurag4DSB — CHANGES_REQUESTED** [review]
Everything looks good. The "OSIS integration with vCloud Director OSE" part is awesome!

Just needs some links for documentation to be referenced for the pre-requisites.
We are not delivering Milestone 1 with S3 Connector and we can expect only dev teams to know how to run individual services. As this documentation audience will be GS/PM along with dev, it needs more references like links to proper documentation which has exact commands that need to be run and which branches to use).

File: docs/milestone-1-validation.md
Comment (anurag4DSB) [reply docs/milestone-1-validation.md:8]: Vault: https://github.com/scality/Vault/blob/development/7.10/TESTING.md#how-to-run-1
Cloudserver: https://github.com/scality/cloudserver/tree/development/7.10#installation
First vault needs to be started.
Only then Cloudserver needs to be started with env variable: `S3VAULT=scality` so that it can use vault as auth backend.

We do not need to mention exact commands over in this doc.
Exact commands to rclone and run vault and then cloudserver will go in the ticket: https://scality.atlassian.net/browse/S3C-4253

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:16]: As the documentation is in a google docs document, I have not mentioned it. I can add it on https://scality.atlassian.net/browse/S3C-4253

**Review by ilkescality — APPROVED** [review]

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:14]: Updated the same in c32406aabea800b30939886856300f757611a70d

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md]: Updated the same in c32406aabea800b30939886856300f757611a70d

File: README.md
Comment (miniscruff) [inline README.md:15]: Although this isn't wrong it feels weird that the first step is to pull from a registry instead of building locally. Why not just `docker build .` ( or similar ).

File: README.md
Comment (miniscruff) [inline README.md:24]: I would simply go over how to run it in dev mode, then at the end include any caveats for production, if any.

File: README.md
Comment (miniscruff) [inline README.md:33]: Since the only difference here is the path to the image, I would just assume the local build instead.

File: README.md
Comment (miniscruff) [inline README.md:46]: Use multiple lines for commands this long

File: README.md
Comment (miniscruff) [inline README.md:50]: Multiline please

File: docs/milestone-1-validation.md
Comment (miniscruff) [inline docs/milestone-1-validation.md:10]: I would not link to the 7.10 version specifically, especially considering the PR I created that includes that env var is not even pushed. Preferably we would just link to the generic docs page but maybe @wabernatScality will have another idea.

**Review by miniscruff — CHANGES_REQUESTED** [review]

File: docs/milestone-1-validation.md
Comment (miniscruff) [inline docs/milestone-1-validation.md:1]: I kinda feel like this file belongs more in citadel than here but can not say for sure.
My reasoning here is that, validating milestones are stepping points in the overall development and may be overwritten or irrelevant by future milestones or releases. Including a doc here feels more permanent. In fact, I would include a warning or note that specifies the document lifecycle is really only related to one milestone.

File: docs/milestone-1-validation.md
Comment (miniscruff) [inline docs/milestone-1-validation.md:8]: Same as below, linking to a specific file in a git repo feels wrong considering how often our branches change (@tcarmet deletes lots of branches)

File: docs/milestone-1-validation.md
Comment (miniscruff) [reply docs/milestone-1-validation.md:10]: I realize now I kinda contradicted myself by saying not to use direct file links as they are not stable while also saying this file feels more like a temporary thing than permanent...
So I guess ignore that if you want but something to potentially consider. :smirk:

File: docs/milestone-1-validation.md
Comment (miniscruff) [inline docs/milestone-1-validation.md:44]: Did you intentionally show a bad secret for the example or what is the point of this?

File: README.md
Comment (saitharunthodupunuru) [reply README.md:15]: For "Running the Application" idea here is to get the docker image generated by eve instead of manually building a docker image. I can add `docker build .` command to Developers guide to generate a docker image locally.

File: README.md
Comment (saitharunthodupunuru) [reply README.md:24]: That is totally true for the developer but for some one, who is just running the application, they don't need to run the dev version unless they want to run unreleased image.

File: README.md
Comment (saitharunthodupunuru) [reply README.md:33]: Production image has a release tag instead of commit hash and release tags will be only made when release ready. Please correct me if I misunderstood the ask here.

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:10]: As this is is only for milestone-1 and milestone-1 should ship with 7.10., this should be good

File: README.md
Comment (miniscruff) [reply README.md:15]: Hmm, I feel like that only works if the docker run command is quick and simple. Like how `docker run mongo` just works. Once you require knowing a git hash or some vague released version it almost becomes a burden. To where, `docker build .` will work every time, plus or minus some extra configs that should also be constant.

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: fixed in 805d047c1c1b2905bf0eca36d088eb40960b1f87

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: fixed in 805d047c1c1b2905bf0eca36d088eb40960b1f87

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:44]: Those are the Vault Super admin credentials and if our Vault super admin secret key does not meet the complexity criteria we should still continue as we must provide Vault super admin credentials here.

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:8]: Same as I explained above, milestone-1 is releasing with 7.10. So for this document, this should be fine

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:1]: Actually this document is accompanying the other milestone-1 document on GIT, https://github.com/scality/vmware-ose-scality/blob/development/1.0/docs/milestone-1-requirements.md

File: docs/milestone-1-validation.md
Comment (miniscruff) [reply docs/milestone-1-validation.md:44]: I feel like the term "criteria" is wrong then, maybe "recommendation". If something said I do not meet the criteria I would assume that meant I couldn't do something and had to try something else, not just hit the ok button anyway.

File: README.md
Comment (miniscruff) [inline README.md:39]: Missed a `-v` after the port. Also I think you need a space after `-p`

File: README.md
Comment (miniscruff) [reply README.md:33]: Yes, correct, but from an outside perspective trying this project for the first time, this line becomes very hard to use.

I can not copy-paste it, as it is missing a key piece of information I can not get from the readme alone. 
By using the local build, the readme is now complete without external information. Alternatively, if the `latest` tag was used such that the version was irrelevant that would work. But, we do not use the latest tag ( or any other generic terms like stable or prod ). 

My point is, without looking up extra information I can not use this step and part of that is to access eve, which is private or existing knowledge of the latest release tags that have been built.

File: README.md
Comment (miniscruff) [inline README.md:35]: Missing space after `-p` and also I think we should use the `--mount` flag instead of `-v`. From here https://docs.docker.com/storage/volumes/#choose-the--v-or---mount-flag it seems the mount flag is pushed more.

File: README.md
Comment (miniscruff) [inline README.md:49]: I would split after each line and add spaces, it works fine to copy paste and breaks up the input more when viewing it.
Same below if you agree.

```suggestion
keytool \
  -genkeypair \
  -keyalg RSA \
  -alias <alias_name> \
  -keystore <key-store_file>.p12 \
  -storepass <key-store_password> \
  -validity <days> \
  -keysize <size_in_bytes> \
  -dname "CN=<common_name>, OU=<organization_unit>, O=<organization>, L=<city>, ST=<state>, C=<country>" \
  -storetype pkcs12
```

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:44]: Correct but that is the error from Vnware OSE, which is not our product. These are only steps to integrate OSE with out product OSIS

Comment (miniscruff) [reply docs/milestone-1-validation.md:44]: Whelp, can't object to that then lol

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md:1]: Hmm that sounds like a good idea. As the other file was created by Terry I did not think of renaming it and adding these details. Please let me know if you still believe it is better to have all details in same file

Comment (miniscruff) [reply docs/milestone-1-validation.md:1]: Yea I would do that, but I would still approve the PR even if you chose not to.

File: README.md
Comment (saitharunthodupunuru) [reply README.md:33]: Sure I will add extra step for `docker build .` to build a locL image. Regarding release tags, anyone should be able to find the release tags on GIT and they can use that right. Correct me if my understanding of generating release tags is different

Comment (saitharunthodupunuru) [reply README.md:33]: FYI there is another PR in progress for release process

https://github.com/scality/vmware-ose-scality/pull/26

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: Completed this in e3ae684ae588760d0b37f12fe614eda2c282a644

File: docs/milestone-1-validation.md
Comment (saitharunthodupunuru) [reply docs/milestone-1-validation.md]: Completed this in e3ae684ae588760d0b37f12fe614eda2c282a644

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: Completed this in e3ae684ae588760d0b37f12fe614eda2c282a644

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: Covered this in e3ae684ae588760d0b37f12fe614eda2c282a644

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: Covered this in e3ae684ae588760d0b37f12fe614eda2c282a644

File: docs/milestone-1.md
Comment (miniscruff) [inline docs/milestone-1.md:43]: Not a deal-breaker but it is weird the ordered list here has new lines between but not others. If we had a markdown linter it might complain one way or the other, I don't quite remember which one was preferred.

File: README.md
Comment (miniscruff) [inline README.md:53]: I don't believe our registry has public read access or any other way to browse available images unlike docker hub. Ideally a short sentence of something like: "You can also use our public registry to run by viewing available images here" with a link to something like registry.scality.com/browse but that doesn't exist. @tcarmet does Harbor provide read-only views of public namespaces? So we could link something like the above?

I just wouldn't repeat the same command 3 times. Plus, it would be nice to provide a means of finding out what the latest images are.

File: README.md
Comment (miniscruff) [reply README.md]: In theory yes, a git tag would map to a docker image but that is purely by convention and without a docker hub tag list or similar seeing the "actual" tags is different. For example: say git tags include a prefix 'v' but docker images don't for whatever reason. My point being, it is just much easier to see the list of actual values over the assumed values from another source. Outside the scope of this issue tho so I will ignore it.

**Review by miniscruff — APPROVED** [review]
Looks good to me for now, we can always come back and adjust the readme and stuff later.

**Comment by saitharunthodupunuru** [comment]
Squashed the commits before merge

File: docs/milestone-1.md
Comment (saitharunthodupunuru) [reply docs/milestone-1.md:43]: corrected the indentation to get ordered list right

**Comment by wabernatScality** [comment]
On documentation files such as this README, it would be a good idea to let me/ask me to look them over before you close out the ticket.

**Comment by saitharunthodupunuru** [comment]
> On documentation files such as this README, it would be a good idea to let me/ask me to look them over before you close out the ticket.

Hey @wabernatScality 

I am sorry. I will add you as a mandatory reviewer for all the subsequent PRs on documentation.
Even though it is merged, I request you to review this PR and give your comments. I can create another PR with the requested changes by you.

Thanks
Sai

**Comment by wabernatScality** [comment]
No worries, Sai!
Just keep me in mind.

**Pull request merged**
